### PR TITLE
[core] specify which chain to create shop

### PR DESCRIPTION
**Pull request by david30907d**
LIQ-1211

## Types of changes

- **New feature**

## Steps to Test This Pull Request

1. update the record from chain=MUMAI to `GOERLI` in your staging db
2. fire the request in the SDK
Screen Shot 2023-01-19 at 12 05 38 PM

### CODE REVIEW

File: core/sdk/src/vendor/config.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { Blockchain } from '@liqnft/candy-shop-types';
 import axios, { AxiosRequestConfig } from 'axios';
 
-const BACKEND_STAGING_URL = 'https://ckaho.liqnft.com/api';
-const BACKEND_PROD_URL = 'https://candy-shop.liqnft.com/api';
+const DATASERVICE_STAGING_URL = 'https://ckaho.liqnft.com/api';
+const DATASERVICE_PROD_URL = 'https://candy-shop.liqnft.com/api';
```
Comment (david30907d) [inline core/sdk/src/vendor/config.ts:5]: seems to me that it's `dataservice` not `backend`

File: core/sdk/src/shop/eth/EthCandyShop.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export class EthCandyShop extends BaseShop {
 
     // Fetch required details for EVM setup
     const shopDetail = await safeAwait(
-      fetchShopsByIdentifier(params.shopCreatorAddress, params.treasuryMint, params.programId)
+      fetchShopsByIdentifier(params.shopCreatorAddress, params.treasuryMint, params.programId, params.env)
```
Comment (david30907d) [inline core/sdk/src/shop/eth/EthCandyShop.ts:41]: per discussion, there's already `env` in `BaseShopConstructorParams`
therefore pass this one as `blockchain` param

File: core/sdk/src/shop/sol/CandyShop.ts [hunk 3]
```diff
@@ -121,7 +121,7 @@ export class CandyShop extends BaseShop implements CandyShopAuctioneer, CandySho
 
     // Fetch required details for SOL setup
     const shopDetail = await safeAwait(
-      fetchShopsByIdentifier(params.shopCreatorAddress, params.treasuryMint, params.programId)
+      fetchShopsByIdentifier(params.shopCreatorAddress, params.treasuryMint, params.programId, params.env)
```
Comment (david30907d) [inline core/sdk/src/shop/sol/CandyShop.ts:124]: ditto

File: core/sdk/src/factory/backend/ShopAPI.ts [hunk 4]
```diff
@@ -42,9 +43,11 @@ export async function fetchShopByIdentifier(
   axiosInstance: AxiosInstance,
   ownerAddress: string,
   mint: string,
-  programId: string
+  programId: string,
+  blockchain: Blockchain
 ): Promise<SingleBase<CandyShop>> {
-  const url = `/shop/owner/${ownerAddress}/mint/${mint}/programId/${programId}`;
+  const mappedBlockchain = mapToCompatibleBlockchain(blockchain);
+  const url = `/shop/owner/${ownerAddress}/mint/${mint}/programId/${programId}/blockchain/${mappedBlockchain}`;
```
Comment (david30907d) [inline core/sdk/src/factory/backend/ShopAPI.ts:50]: per discussion, extracted a mapper logic into utils

File: core/sdk/src/factory/backend/OrderAPI.ts [hunk 5]
```diff
@@ -30,14 +29,7 @@ export async function fetchOrdersByStoreId(
...
   } = ordersFilterQuery;
-
-  let blockchain = ordersFilterQuery.blockchain;
-  if (
-    ordersFilterQuery.blockchain === Blockchain.SolDevnet ||
-    ordersFilterQuery.blockchain === Blockchain.SolMainnetBeta
-  ) {
-    blockchain = Blockchain.Sol;
-  }
+  const blockchain = mapToCompatibleBlockchain(ordersFilterQuery.blockchain);
```
Comment (david30907d) [inline core/sdk/src/factory/backend/OrderAPI.ts:32]: ditto

File: core/sdk/src/vendor/config.ts (hunk 1 above)
Comment (RustySol) [reply core/sdk/src/vendor/config.ts:5]: let's keep the `backend` naming

Comment (david30907d) [reply core/sdk/src/vendor/config.ts:5]: got it~

**Review by RustySol — APPROVED** [review]

**Pull request merged**